### PR TITLE
fix: update publish workflow

### PR DESCRIPTION
**Pull request by CraigCottingham**
- rework workflow to match what we’re doing in other workflows
- remove storybook and deploy stages from .travis.yml

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,70 +1,61 @@
...
   publish_packages:
-    runs-on: 'ubuntu-latest'
+    if: github.event.action == 'closed' && github.event.pull_request.merged
+    runs-on: ubuntu-latest
     steps:
-      - name: 'clone repo with full history for lerna'
+      - name: check out latest commit
         uses: actions/checkout@v2
         with:
-          fetch-depth: 0
```
Comment (kaborso) [inline .github/workflows/publish.yml:12]: Is `lerna` able to version without this setting?

Comment (kaborso) [reply .github/workflows/publish.yml:12]: Can we do an example run with `on: pull_request(synchronize)`?

Comment (CraigCottingham) [reply .github/workflows/publish.yml:12]: Beats me. I know nothing about `lerna`. I _do_ know that packages are published when this workflow is run. I _don't_ know if that one example is sufficiently comprehensive.

Comment (CraigCottingham) [reply .github/workflows/publish.yml:12]: Regarding a test run: I commented out the `on` criteria and the `if` condition on the `publish_packages` job, and [packages published as expected](https://github.com/rentpath/react-ui/packages).

Comment (kaborso) [reply .github/workflows/publish.yml:12]: Awesome

**Review by kaborso — APPROVED** [review]

**Review by tadjohnston — APPROVED** [review]

**Pull request merged**